posts/deploying-a-jhipster-app-to-aws-using-elastic-beanstalk.md

```diff
@@ -0,0 +1,101 @@
+    "serverPort": "8080",
+    "authenticationType": "jwt",
+    "cacheProvider": "ehcache",
+    "enableHibernateCache": true,
+    "websocket": false,
+    "databaseType": "sql",
+    "devDatabaseType": "h2Disk",
+    "prodDatabaseType": "postgresql",
+    "searchEngine": false,
+    "messageBroker": false,
+    "serviceDiscoveryType": false,
+    "buildTool": "maven",
+    "enableSwaggerCodegen": false,
+    "embeddableLaunchScript": false,
+    "useSass": true,
+    "clientPackageManager": "npm",
+    "clientFramework": "react",
+    "clientTheme": "none",
+    "clientThemeVariant": "",
+    "testFrameworks": [],
+    "jhiPrefix": "jhi",
+    "entitySuffix": "",
+    "dtoSuffix": "DTO",
+    "otherModules": [],
+    "enableTranslation": true,
+    "nativeLanguage": "en",
+    "languages": [
+      "en"
+    ],
+    "blueprints": []
+  }
+}
+```
+
+The `.yo-rc.json` above can be used to avoid answering all the questions manually.
+
+# Deployment to AWS with Elastic Beanstalk
+
+ First, make sure that your AWS CLI is [correctly configured](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html#cli-quick-configuration) and the account must have the `AWSElasticBeanstalkFullAccess` policy. If you use a named profile different than `default`, you just have to set the environment variable `AWS_PROFILE` with the correct profile.
+
+ The command `jhipster aws` will ask questions about the AWS deployment:
+
+ ![](https://raw.githubusercontent.com/falydoor/blog-usa/jhipster-on-aws-with-beanstalk/images/2019/11/jhipster-aws.png)
+
+ The AWS account used needs access to the S3 bucket since it will be used as a storage location for the application's Jars. The sub-generator will also create a RDS instance so make sure that the database credentials are correct and match the ones in `application-prod.yml`. The application will always be packaged using the `prod` profile, which means that all the tests must be green!
+
+ Once the Elastic Beanstalk application is created, navigate to the [Elastic Beanstalk console](https://console.aws.amazon.com/elasticbeanstalk/home) to view the details of the application. Use the public URL to check that everything was correctly deployed and running.
+
+ ![](https://raw.githubusercontent.com/falydoor/blog-usa/jhipster-on-aws-with-beanstalk/images/2019/11/eb-url.png)
+
+ If the deployment failed, the logs page can be used to view all the logs of the application. The logs from `/var/log/tomcat8/catalina.out` are the JHipster logs, those should be used to find out why the application did not start.
+
+# Elastic Beanstalk application improvements
+
+By default, your Elastic Beanstalk application uses a classic load balancer with auto scaling up to 4 instances. The default configuration is enough for demo purpose, but if you expect a heavy load I recommend updating the configuration. Also, it might be a better idea to set the scaling trigger to the `CPUUtilization` metric instead of `NetworkOut`.
```

Comment:
```suggestion
By default, your Elastic Beanstalk application uses a classic load balancer with auto-scaling up to 4 instances. The default configuration is enough for demo purposes, but if you expect heavier traffic update the configuration for more instances. Also, it might be a better idea to set the scaling trigger to the `CPUUtilization` metric instead of `NetworkOut`.
```